kernel/data-pipeline/dialect/opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/wal/decode/MppdbDecodingPlugin.java

```diff
@@ -59,30 +68,46 @@ public final class MppdbDecodingPlugin implements DecodingPlugin {
     
     private final BaseTimestampUtils timestampUtils;
     
+    private final boolean decodeWithXid;
+    
+    public MppdbDecodingPlugin(final BaseTimestampUtils timestampUtils) {
+        this.timestampUtils = timestampUtils;
+        decodeWithXid = false;
+    }
+    
     @Override
     public AbstractWALEvent decode(final ByteBuffer data, final BaseLogSequenceNumber logSequenceNumber) {
         AbstractWALEvent result;
-        char eventType = readOneChar(data);
-        result = '{' == eventType ? readTableEvent(readMppData(data)) : new PlaceholderEvent();
+        String dataText = StandardCharsets.UTF_8.decode(data).toString();
+        if (decodeWithXid) {
+            result = decodeDataWithXid(dataText);
+        } else {
+            result = decodeDataIgnoreXid(dataText);
+        }
         result.setLogSequenceNumber(logSequenceNumber);
         return result;
     }
     
-    private char readOneChar(final ByteBuffer data) {
-        return (char) data.get();
+    private AbstractWALEvent decodeDataWithXid(final String dataText) {
+        AbstractWALEvent result;
+        if (dataText.startsWith("{")) {
+            result = readTableEvent(dataText);
+            return result;
+        }
+        Matcher beginXidMatcher = PATTERN_BEGIN_XID.matcher(dataText);
+        Matcher commitXidMatcher = PATTERN_COMMIT_XID.matcher(dataText);
```

Comment:
It might hurt performance when pattern match frequently here, it's better to use streaming parsing at one time